ecommerce/cybersource_api.py

```diff
@@ -0,0 +1,99 @@
+"""Wrapper class for CyberSource"""
+
+import json
+import logging
+
+from django.db import transaction
+
+from ecommerce.models import Order, FulfilledOrder, Transaction
+from main.settings import ECOMMERCE_DEFAULT_PAYMENT_GATEWAY
+from ecommerce.constants import TRANSACTION_TYPE_REFUND, REFUND_SUCCESS_STATES
+from mitol.payment_gateway.api import (
+    Refund as GatewayRefund,
+    PaymentGateway,
+)
+from mitol.payment_gateway.exceptions import RefundDuplicateException
+
+log = logging.getLogger(__name__)
+
+
+def refund_order(*, order_id: int, **kwargs):
+    """
+    A function that performs refund for a given order id
+
+    Args:
+       order_id (int): Id of the order which is being refunded
+       kwargs (dict): Dictionary of the other attributes that are passed e.g. refund amount, refund reason, unenroll
+       If no refund_amount is provided it will use refund amount from Transaction obj
+       unenroll will never be performed if the refund fails
+
+    Returns:
+        tuple(bool, bool):
+        First bool identifies if the refund operation was a success or failure
+        Second bool identifies if unrenroll was success or failure
+    """
+    refund_amount = kwargs.get("refund_amount")
+    refund_reason = kwargs.get("refund_reason", "")
+    unenroll = kwargs.get("unenroll", False)
+
+    with transaction.atomic():
+        # Get order details for the given order_id (Optimally, This should be a fulfilled order)
+        order = FulfilledOrder.objects.select_for_update().get(id=order_id)
+
+        if order.state != Order.STATE.FULFILLED:
+            log.debug(f"Order with order_id {order_id} is not in fulfilled state.")
+            return False, False
+
+        # Extract transaction_id from order CyberSource json
+        order_recent_transaction = Transaction.objects.filter(order=order_id).first()
+
+        if not order_recent_transaction:
+            log.error(
+                f"There is no associated transaction against order_id {order_id}."
+            )
+            return False, False
+
+        transaction_dict = json.loads(json.dumps(order_recent_transaction.data))
```

Comment:
Could you explain why you're cycling this through json?

Reply:
It shouldn't be, I initially had some bad data in the transaction somehow, maybe because I created some orders through the factory and that wasn't loading correctly as JSON as is, so I had to cycle it but I'm sure it can be removed and we can use `transaction.data` as is. Will be done as part of the feedback.